### PR TITLE
[bfa]  CreatureFamily struct update 

### PR DESCRIPTION
**Pull request by Palabola**
**Changes proposed:**

-  New BFA dbc values

**Target branch(es):** 3.3.5/master

- [x] master

**Issues addressed:** Closes #  (insert issue tracker number)

**Tests performed:** (Does it build, tested in-game, etc.)

**Known issues and TODO list:** (add/remove lines as needed)

### CODE REVIEW

**Review by Shauren — CHANGES_REQUESTED** [review]
You must change .Get*** function used to read that field when parsing all involved QueryResults

**Review by Shauren — CHANGES_REQUESTED** [review]

**Comment by illfated** [comment]
BTW, please add a LineFeed or `<Enter>` at the end of `sql/updates/world/master/2018_10_08_00_world.sql` (missing NewLine).

File: src/server/game/Miscellaneous/SharedDefines.h [hunk 1]
```diff
@@ -3977,7 +3977,17 @@ enum CreatureFamily
     CREATURE_FAMILY_ABYSSAL             = 148,
     CREATURE_FAMILY_RYLAK               = 149,
     CREATURE_FAMILY_RIVERBEAST          = 150,
-    CREATURE_FAMILY_STAG                = 151
+    CREATURE_FAMILY_STAG                = 151,
+    CREATURE_FAMILY_MECHANICAL          = 154,
+    CREATURE_FAMILY_ABONIATION          = 155,
```
[the remark this answers is not in the document]
Comment (illfated) [reply src/server/game/Miscellaneous/SharedDefines.h]: Good point. `CREATURE_FAMILY_ABOMINATION` should be the correct name, unless there is a sniff sample showing that Blizzard have actually added this typo in their servers.
Especially if this is a new enum name, using a correct spelling will be more practical.
(I tried searching for `CREATURE_FAMILY_ABONIATION` as well as `CREATURE_FAMILY_ABOMINATION` in both master & 3.3.5, but could not find it being used before.)

[the remark this answers is not in the document]
Comment (Palabola) [reply src/server/game/Miscellaneous/SharedDefines.h]: Typo

File: src/server/game/Miscellaneous/SharedDefines.h [hunk 2]
```diff
@@ -3977,7 +3977,17 @@ enum CreatureFamily
     CREATURE_FAMILY_ABYSSAL             = 148,
     CREATURE_FAMILY_RYLAK               = 149,
     CREATURE_FAMILY_RIVERBEAST          = 150,
-    CREATURE_FAMILY_STAG                = 151
+    CREATURE_FAMILY_STAG                = 151,
+    CREATURE_FAMILY_MECHANICAL          = 154,
+    CREATURE_FAMILY_ABOMINATION         = 155,
+    CREATURE_FAMILY_SCALEHIDE           = 156,
+    CREATURE_FAMILY_OXEN                = 157,
+    CREATURE_FAMILY_FEATHEMANE          = 160,
```
Comment (arazi47) [inline src/server/game/Miscellaneous/SharedDefines.h]: CREATURE_FAMILY_FEATHERMANE.

File: src/server/game/Server/Packets/CalendarPackets.cpp [hunk 3]
```diff
@@ -121,34 +141,42 @@ void WorldPackets::Calendar::CalendarAddEvent::Read()
...
-    uint8 titleLen = _worldPacket.ReadBits(8);
-    uint16 descLen = _worldPacket.ReadBits(11);
+    uint8 titleLen = buffer.ReadBits(8);
+    uint16 descLen = buffer.ReadBits(11);
 
-    EventInfo.Title = _worldPacket.ReadString(titleLen);
-    EventInfo.Description = _worldPacket.ReadString(descLen);
+    updateEventInfo.Title = buffer.ReadString(titleLen);
+    updateEventInfo.Description = buffer.ReadString(descLen);
+}
```
Comment (illfated) [inline src/server/game/Server/Packets/CalendarPackets.cpp]: from travis-ci :
```cpp
src/server/game/Server/Packets/CalendarPackets.cpp:159:1:
 fatal error: control reaches end of non-void function [-Wreturn-type]
}
^
```

File: src/server/game/Server/Protocol/Opcodes.cpp [hunk 4]
```diff
@@ -874,83 +874,83 @@ void OpcodeTable::Initialize()
...
     DEFINE_SERVER_OPCODE_HANDLER(SMSG_ACCOUNT_DATA_TIMES,                      STATUS_NEVER,        CONNECTION_TYPE_REALM);
     DEFINE_SERVER_OPCODE_HANDLER(SMSG_ACCOUNT_HEIRLOOM_UPDATE,                 STATUS_NEVER,        CONNECTION_TYPE_INSTANCE);
-    DEFINE_SERVER_OPCODE_HANDLER(SMSG_ACCOUNT_MOUNT_UPDATE,                    STATUS_NEVER,        CONNECTION_TYPE_INSTANCE);
-    DEFINE_SERVER_OPCODE_HANDLER(SMSG_ACCOUNT_TOYS_UPDATE,                     STATUS_NEVER,        CONNECTION_TYPE_INSTANCE);
-    DEFINE_SERVER_OPCODE_HANDLER(SMSG_ACHIEVEMENT_DELETED,                     STATUS_NEVER,        CONNECTION_TYPE_INSTANCE);
-    DEFINE_SERVER_OPCODE_HANDLER(SMSG_ACHIEVEMENT_EARNED,                      STATUS_NEVER,        CONNECTION_TYPE_INSTANCE);
-    DEFINE_SERVER_OPCODE_HANDLER(SMSG_ACTIVATE_TAXI_REPLY,                     STATUS_NEVER,        CONNECTION_TYPE_REALM);
-    DEFINE_SERVER_OPCODE_HANDLER(SMSG_ACTIVE_GLYPHS,                           STATUS_NEVER,        CONNECTION_TYPE_REALM);
+    DEFINE_SERVER_OPCODE_HANDLER(SMSG_ACCOUNT_MOUNT_UPDATE,                    XXXXXXXXXXXX,        CONNECTION_TYPE_INSTANCE);
```
Comment (illfated) [inline src/server/game/Server/Protocol/Opcodes.cpp]: from travis-ci :
```cpp
src/server/game/Server/Protocol/Opcodes.cpp:877:80:
 fatal error: use of undeclared identifier 'XXXXXXXXXXXX'
    DEFINE_SERVER_OPCODE_HANDLER(SMSG_ACCOUNT_MOUNT_UPDATE,                    XXXXXXXXXXXX,        CONNECTION_TYPE_INSTANCE);
                                                                               ^
```

**Pull request closed without merging**